File: Behavioral/Visitor/Cpp/ExportVisitor.cpp
// ExportVisitor.cpp

#include "ExportVisitor.h"
#include "Circle.h"
#include "Square.h"

void ExportVisitor::visit(Circle& circle) {
  std::cout << "Exporting a Circle to file." << std::endl;
}

void ExportVisitor::visit(Square& square) {
  std::cout << "Exporting a Square to file." << std::endl;
}

File: Behavioral/Visitor/CSharp/ExportVisitor.cs
// ExportVisitor.cs

using System;

public class ExportVisitor : IShapeVisitor {
  public void Visit(Circle circle) {
    Console.WriteLine("Exporting a Circle to file.");
  }

  public void Visit(Square square) {
    Console.WriteLine("Exporting a Square to file.");
  }
}
